ql/src/java/org/apache/hadoop/hive/ql/parse/MergeSemanticAnalyzer.java

```diff
@@ -441,6 +442,11 @@ private String handleUpdate(ASTNode whenMatchedUpdateClause, StringBuilder rewri
         default:
           //do nothing
         }
+
+        if ("`default`".equalsIgnoreCase(rhsExp.trim())) {
+          rhsExp = MapUtils.getString(colNameToDefaultConstraint, name, "null");
```

Comment:
iiuc this makes changes the column value to the default if we see the "default" in the query

I see that this also work for a plain insert:
```
create table q2(a string default 'asd');
insert into q2 values(`default`)
select * from q2;
```

however I think the standard suggest to use the `DEFAULT` keyword and not as a string literal; in Hive we seem to also "support" the "default" as a string literal to be interpreted as default.

I think because of various reasons - the default keyword becomes the string default at some point and it works like that right now.

could you open a follow-up to fix the `default` literal's handling?